jaeger-context/src/main/java/com/uber/jaeger/context/TracingUtils.java

```diff
@@ -14,18 +14,28 @@
 
 package com.uber.jaeger.context;
 
+import io.opentracing.util.GlobalTracer;
+
 import java.util.concurrent.ExecutorService;
 
 public class TracingUtils {
-  private static final TraceContext traceContext = new ThreadLocalTraceContext();
+  private static TraceContext traceContext = new ActiveSpanSourceTraceContext();
 
   public static TraceContext getTraceContext() {
+    assertGlobalTracerIsRegistered();
     return traceContext;
   }
 
   public static ExecutorService tracedExecutor(ExecutorService wrappedExecutorService) {
+    assertGlobalTracerIsRegistered();
     return new TracedExecutorService(wrappedExecutorService, traceContext);
   }
 
-  private TracingUtils(){}
+  private static void assertGlobalTracerIsRegistered() {
+    if (!GlobalTracer.isRegistered()) {
+      throw new IllegalStateException("Please register a io.opentracing.util.GlobalTracer.");
```

Comment:
@yurishkuro I felt it was simpler to use `GlobalTracer.get()` directly in `ActiveSpanSourceTraceContext`. It makes tests dirtier, but it's temporary code. 
What is your opinion on the `assertGlobalTracerIsRegistered` check?

Reply:
why not pass it the GlobalTracer.INSTANCE as we discussed? By using GlobalTracer directly you're introducing tight coupling in the new class, which didn't need to happen.

Reply:
The type of `GlobalTracer.INSTANCE` is `GlobalTracer` and I didn't see that as being particularly advantageous because it still couples `ActiveSpanSourceTraceContext` to the implementation of `GlobalTracer`.